core/autoscaling.go

```diff
@@ -88,18 +117,78 @@ func (a *autoScalingGroup) loadConfOnDemand() bool {
 					return done
 				}
 			}
+		}
+		debug.Println("Couldn't find tag", tagKey)
+	}
+	return false
+}
+
+func (a *autoScalingGroup) loadBiddingPolicy(tagValue *string) (string, bool) {
+	biddingPolicy := *tagValue
+	if biddingPolicy != "aggressive" {
+		return DefaultBiddingPolicy, false
+	}
+
+	logger.Printf("Loaded BiddingPolicy value with %s from tag %s\n", biddingPolicy, BiddingPolicyTag)
+	return biddingPolicy, true
+}
+
+func (a *autoScalingGroup) loadConfSpot() bool {
+	tagList := []string{BiddingPolicyTag}
+	loadDyn := map[string]func(*string) (string, bool){
+		BiddingPolicyTag: a.loadBiddingPolicy,
+	}
+
+	for _, tagKey := range tagList {
+		if tagValue := a.getTagValue(tagKey); tagValue != nil {
+			if _, ok := loadDyn[tagKey]; ok {
+				if newValue, done := loadDyn[tagKey](tagValue); done {
+					a.region.conf.BiddingPolicy = newValue
+					logger.Println("BiddingPolicy =", a.region.conf.BiddingPolicy)
+					return done
+				}
+			}
 		} else {
 			debug.Println("Couldn't find tag", tagKey)
 		}
 	}
 	return false
 }
```

Comment:
Applying the same principle as before, it could be re-written as:
```go
func (a *autoScalingGroup) loadConfSpot() bool {
	tagList := []string{BiddingPolicyTag}
	loadDyn := map[string]func(*string) (string, bool){
		BiddingPolicyTag: a.loadBiddingPolicy,
	}

	for _, tagKey := range tagList {
		tagValue := a.getTagValue(tagKey)
		if tagValue == nil {
			debug.Println("Couldn't find tag", tagKey)
			continue
		}
		if _, ok := loadDyn[tagKey]; !ok {
			debug.Println("No method associated with tag", tagKey)
			continue
		}
		if newValue, done := loadDyn[tagKey](tagValue); done {
			a.region.conf.BiddingPolicy = newValue
			logger.Println("BiddingPolicy =", a.region.conf.BiddingPolicy)
			return done
		}
	}
	return false
}
```

Or considering we have only one policy, even as simple as:
```go
func (a *autoScalingGroup) loadConfSpot() bool {
	tagValue := a.getTagValue(BiddingPolicyTag)
	if tagValue == nil {
		debug.Println("Couldn't find tag", BiddingPolicyTag)
		return false
	}
	if newValue, done := a.loadBiddingPolicy(tagValue); done {
		a.region.conf.BiddingPolicy = newValue
		logger.Println("BiddingPolicy =", a.region.conf.BiddingPolicy)
		return done
	}
	return false
}
```

I presume both could help/should solve the codeClimate issue

Reply:
The first one still failed the CodeClimate run.... So, I used the second one.